.github/workflows/mypy.yaml

```diff
@@ -1,6 +1,6 @@
 name: MyPy Lint
 on:
-  pull_request:
+  push
```

Comment:
I'm not certain if we need to lint every push. For instance, checking seems unnecessary when we push tags.
From what I understand, the `pull_request` trigger will initiate the action each time we push to the branch associated with the pull request.